cmd/cigarToBed/cigarToBed.go

```diff
@@ -0,0 +1,198 @@
+	fmt.Println(visualize)
+
+	//raven added this block to put visualized alignment output into MSA Fasta
+	if outFa != "" { //only write to file if a filename is given in the faOut command line option
+		outFaFile, err := os.Create(outFa) //fileio.EasyCreate and other fileio tools should work, but I don't really know how to use them
+		if err != nil {
+			log.Fatalf("Write to file failed on step 1\n")
+		}
+		visualizeOutput := ">" + faOne.Name + "\n" + strings.Split(visualize, "\n")[0] + "\n" + ">" + faTwo.Name + "\n" + strings.Split(visualize, "\n")[1] + "\n"
+		_, err = outFaFile.WriteString(visualizeOutput)
+		if err != nil {
+			log.Fatalf("Write to file failed on step 2\n")
+		}
+		outFaFile.Close() //commented out defer close
+	}
+
+	//cigar to graph
+	//genomeGraph := cigarToGraph(faOne, faTwo, aln)
+	//genomeGraph.PrintGraph(genomeGraph)
+}
+
+//raven edited this block to specify only 1 sequnce is expected in each fasta file and add Usage nad options
+func usage() {
+	fmt.Print(
+		"./cigarToBed\n" +
+			" Uses globalAlignment, affineGap (instead of constGap) to align 2 .fasta files, each with only 1 sequence, then convert cigars to ins and del beds\n" +
+			"Usage:\n" +
+			" cigartobed target.fasta query.fasta\n" +
+			"options:\n" +
+			"	-faOut=fasta MSA output filename\n" +
+			" -insBedOut=insertion bed output filename\n" +
+			" -delBedOut=deletion bed output filename\n" +
+			" -FirstPos_Ins=first base position of the query sequence, which will be used to make the insertion bed\n" +
+			" -FirstPos_Del=first base position of the target sequence, which will be used to make the deletion bed\n" +
+			" -Chr=chromosome name\n")
+	flag.PrintDefaults()
+}
+
+func main() {
+	var expectedNum int = 2
+
+	flag.Usage = usage
+	log.SetFlags(log.Ldate | log.Ltime | log.Lshortfile)
+	faOut := flag.String("faOut", "", "name of the MSA output file") //raven added this line
+	insBedOut := flag.String("insBedOut", "ins.bed", "insertion bed output filename")
+	delBedOut := flag.String("delBedOut", "del.bed", "deletion bed output filename")
+	FirstPos_Ins := flag.Int("FirstPos_Ins", 1, "first base position of the query sequence, which will be used to make the insertion bed")
+	FirstPos_Del := flag.Int("FirstPos_Del", 1, "first base position of the target sequence, which will be used to make the deletion bed")
+	Chr := flag.String("Chr", "chr1", "chromosome name")
+	flag.Parse()
+
+	if len(flag.Args()) != expectedNum {
+		flag.Usage()
+		log.Fatalf("error, expecting 2 .fasta files to be able to align, only found %d files...\n", len(flag.Args()))
+	}
+
+	//read in sequences that should be put in as fasta type files.
+	//raven edited this block to save fileio.EasyOpen as file handles, so that the file is only opened 1 time for 2 purposes: faDone and CountSeqIdx
+	inputFileOne := fileio.EasyOpen(flag.Arg(0)) //raven's note: EasyOpen returns the type EasyReader
+	inputFileTwo := fileio.EasyOpen(flag.Arg(1))
+	outFa := *faOut
```

Comment:
This works fine, though another option would be to de-reference the pointers when you call the CigarToBed function (e.g. `GlobalAlignment_CigarToBed( ... , *faOut, *insBedOut, *delBedOut, ... )`)

Reply:
Just changed this. Now there are much fewer lines. Thank you!